server/src/internalClusterTest/java/org/elasticsearch/cluster/ClusterStateDiffIT.java

```diff
@@ -309,14 +309,16 @@ private IndexRoutingTable randomChangeToIndexRoutingTable(IndexRoutingTable orig
         for (int i = 0; i < original.size(); i++) {
             IndexShardRoutingTable indexShardRoutingTable = original.shard(i);
             Set<String> availableNodes = Sets.newHashSet(nodes);
-            for (ShardRouting shardRouting : indexShardRoutingTable.shards()) {
+            for (int j = 0; j < indexShardRoutingTable.size(); j++) {
```

Comment:
Not convinced I like standardising on `j` for the loop variable here (if I'd known this was coming I would have objected to `i` for the outer loop too). This is going to get copy-pasted everywhere because IntelliJ no longer knows how to auto-generate this loop. Could we use a better variable name in both places?

Reply:
Sure, before I go ahead with that :) : how about `shard` and `copy`?

Reply:
sounds good to me

Reply:
Actually maybe `shardId` over `shard`? `original.shard(shard);` is a bit odd, vs `original.shard(shardId);`